src/quo/components/notifications/count_down_circle.cljs

```diff
@@ -49,60 +48,61 @@
-                                               current-elapsed      (swap! elapsed-time + delta)
-                                               current-display-time (+ start-at current-elapsed)
-                                               completed?           (>= current-display-time duration)]
-                                           (reset! display-time (if completed?
-                                                                  duration
-                                                                  current-display-time))
-                                           (when-not completed?
-                                             (reset! prev-frame-time frame-time)
-                                             (reset! frame-request (js/requestAnimationFrame
-                                                                    swap-elapsed-time-each-frame))))))]
-    (reagent/create-class
-     {:component-will-unmount #(js/cancelAnimationFrame @frame-request)
-      :reagent-render
-      (fn []
-        (reset! frame-request (js/requestAnimationFrame swap-elapsed-time-each-frame))
-        [rn/view
-         {:style {:position :relative
-                  :width    size
-                  :height   size}}
-         [svg/svg
-          {:view-box (str "0 0 " size " " size)
-           :width    size
-           :height   size}
-          [svg/path
-           {:d path :fill :none :stroke (or trail-color :transparent) :stroke-width stroke-width}]
-          (when-not (= @display-time duration)
-            [svg/path
-             {:d                 path
-              :fill              :none
-              :stroke            (or color (get-in themes [:color theme]))
-              :stroke-linecap    :square
-              :stroke-width      stroke-width
-              :stroke-dasharray  path-length
-              :stroke-dashoffset (linear-ease @display-time 0 path-length duration)}])]])})))
+(defn circle-timer-internal
+  [{:keys [color duration size stroke-width trail-color rotation initial-remaining-time]}]
+  (let [theme                           (quo.theme/use-theme-value)
+        rotation                        (or rotation :clockwise)
+        duration                        (or duration 4)
+        stroke-width                    (or stroke-width 1)
+        size                            (or size 9)
+        max-stroke-width                (max stroke-width 0)
+        {:keys [path path-length]}      (get-path-props size max-stroke-width rotation)
+        start-at                        (get-start-at duration initial-remaining-time)
+        elapsed-time                    (rn/use-ref-atom 0)
+        prev-frame-time                 (rn/use-ref-atom nil)
+        frame-request                   (rn/use-ref-atom nil)
+        [display-time set-display-time] (rn/use-state start-at)
+        swap-elapsed-time-each-frame    (fn swap-elapsed-time-each-frame [frame-time]
```

Comment:
so basically this animation blocks the js thread, and I'm not sure it worked differently before, but probably because in reagent we use timer it worked different, but anyway it should be implemented differently, probably with reanimated? any ideas, advice @Parveshdhull @clauxx @J-Son89

Reply:
I'm not sure why requestAnimationFrame is used here

Reply:
`requestAnimationFrame(fn) is not the same as setTimeout(fn, 0) - the former will fire after all the frames have flushed` but why js thread is blocked then

Reply:
Hmm I'm not sure, would have to dig into the code. Could remove that refactor from this p and we handle this one separately?

Reply:
fixed by removing extra frame request cc @J-Son89

Reply:
Nice! 👌